mutiny-core/src/node.rs

```diff
@@ -1105,133 +1102,170 @@ async fn start_reconnection_handling(
-                log_debug!(
-                    reconnection_logger,
-                    "stopping reconnection component for node: {}",
-                    node_pubkey.to_hex(),
-                );
-                stop_component(&reconnection_stopped_components);
-                log_debug!(
-                    reconnection_logger,
-                    "stopped reconnection component for node: {}",
-                    node_pubkey.to_hex(),
-                );
-                break;
+            // run through reconnection logic every delay seconds,
+            // check if it should stop once every second.
+            for _ in 0..delay {
+                if reconnection_stop.load(Ordering::Relaxed) {
+                    log_debug!(
+                        reconnection_logger,
+                        "stopping reconnection component for node: {}",
+                        node_pubkey.to_hex(),
+                    );
+                    stop_component(&reconnection_stopped_components);
+                    log_debug!(
+                        reconnection_logger,
+                        "stopped reconnection component for node: {}",
+                        node_pubkey.to_hex(),
+                    );
+                    return;
+                }
+                sleep(1_000).await;
             }
 
-            if !multi_socket_reconnect.connected() {
+            let connected = { multi_socket_reconnect.lock().await.connected() };
+            if !connected {
                 log_debug!(
                     reconnection_logger,
                     "got disconnected from multi socket proxy, going to reconnect"
                 );
                 match WsProxy::new(
                     &websocket_proxy_addr_copy,
-                    self_connection_copy.clone(),
+                    self_connection.clone(),
                     reconnection_logger.clone(),
                 )
                 .await
                 {
                     Ok(main_proxy) => {
-                        multi_socket_reconnect.reconnect(Arc::new(main_proxy)).await;
+                        multi_socket_reconnect
+                            .lock()
+                            .await
+                            .connect(Arc::new(main_proxy))
+                            .await;
+                        // reset delay to initial value if reconnection successful
+                        delay = 5;
                     }
-                    Err(_) => {
-                        sleep(5 * 1000).await;
-                        continue;
+                    Err(e) => {
+                        log_error!(
+                            reconnection_logger,
+                            "could not create new multi socket proxy: {e}",
+                        );
+                        // double the delay if reconnection fails, but do not exceed 60 seconds
+                        delay = (delay * 2).min(60);
```

Comment:
should make `5` and `60` `const`s for easier reference, these values are used in a bunch of places, would add more context and make them easier to update

Reply:
Good suggestion, added.

Reply:
think you forgot to push